astra/src/main/java/com/slack/astra/metadata/schema/FieldType.java

```diff
@@ -133,7 +136,22 @@ public Schema.SchemaFieldType toSchemaFieldType() {
   DATE("date") {
     @Override
     public void addField(Document doc, String name, Object value, LuceneFieldDef fieldDef) {
-      LONG.addField(doc, name, value, fieldDef);
+      if (value instanceof LocalDateTime localDateTime) {
+        Instant instant = localDateTime.toInstant(ZoneOffset.UTC);
+        long timeSinceEpoch = instant.toEpochMilli();
+        if (fieldDef.isIndexed) {
+          doc.add(new LongPoint(name, timeSinceEpoch));
+        }
+        if (fieldDef.isStored) {
+          doc.add(new StoredField(name, localDateTime.toString()));
+        }
+        if (fieldDef.storeDocValue) {
+          doc.add(new NumericDocValuesField(name, timeSinceEpoch));
+        }
+      } else {
+        // back-compat
+        LONG.addField(doc, name, value, fieldDef);
```

Comment:
so when we deploy the indexer first, docs will keep getting added as this. Once we deploy preprocessor the object will become LocalDateTime